pkg/agent/billing/billing.go

```diff
@@ -265,26 +266,26 @@ func (s *metricsState) drainAppendToBatch(logger *zap.Logger, conf *Config, batc
 	for key, history := range s.historical {
 		history.finalizeCurrentTimeSlice()
 
-		batch.AddIncrementalEvent(logAddedEvent(logger, billing.IncrementalEvent{
+		batch.Add(logAddedEvent(logger, billing.Enrich(batch, &billing.IncrementalEvent{
```

Comment:
I missed this change the first time I read this. I'm not super into this, which is easy enough to miss and ends up being "lets pass batch it itself."

Reply:
Not sure what you mean? Also, have since updated, so maybe that resolves this?

Reply:
while you don't do this, this code sort of looks like you might be... passing (and storing?) batch inside of the Event... I think it's maybe/probably fine, it just sort of looks like a leaky abstraction or a possible circular dependency.